xdsl/ir.py

```diff
@@ -297,6 +297,9 @@ def build(cls: type[A], *args: Any) -> A:
         assert False
 
     def __post_init__(self):
+        self._verify()
+
+    def _verify(self):
```

Comment:
This is backwards. `verify` should be the one calling `verify_`.

Reply:
Also, whis is this name `_verify` and not `verify_` like what is currently being used?

Reply:
`verify_` is not being used on `Attribute`, only on `Operation`. I added a `_verify` with an underscore because it's a private method and should not be overridden except in `ir.py`, to include the pyrdl check

Reply:
Oh I see, because `__post__init__` will call `_verify`, and users won't ever call `_verify` (since `Attribute` always verify).
But then, why having this function in the first place? We should just call `verify` in `__post_init__`?

Reply:
Because I override the private one in the subclass to call the irdl verifier.

Reply:
We could also override `__post_init__`, but that seems a little weird to me